prometheus/labels.go

```diff
@@ -31,6 +31,64 @@ import (
 // create a Desc.
 type Labels map[string]string
 
+// ConstrainedLabels represents a label name and its constrain function
+// to normalize label values. This type is commonly used when constructing
+// metric vector Collectors.
+type ConstrainedLabel struct {
+	Name       string
+	Constraint func(string) string
+}
+
+func (cl ConstrainedLabel) Constrain(v string) string {
+	if cl.Constraint != nil {
+		return cl.Constraint(v)
+	}
+	return v
+}
+
+// ConstrainableLabels is the interface to create label constraints
+type ConstrainableLabels interface {
+	constrainedLabels() ConstrainedLabels
+	labelNames() []string
+}
+
+// ConstrainedLabels represents a collection of label name -> constrain function
+// to normalize label values. This type is commonly used when constructing
+// metric vector Collectors.
+type ConstrainedLabels []ConstrainedLabel
+
+// constrainedLabels implements ConstrainableLabels
```

Comment:
No need for this comment, it's not exported method and nothing useful here: 🙈

Reply:
Removed 🗑️